go/rpcclientlib/README.md

```diff
@@ -1 +1,16 @@
-This package contains library code to allow client applications to connect to Obscuro nodes via RPC.
\ No newline at end of file
+This package contains library code to allow client applications to connect to Obscuro nodes via RPC.
+
+### Viewing keys
+
+Viewing keys generated inside the wallet extension (or other users of the obscuro rpc client), and then signed by the wallet (e.g. MetaMask)
+to which the keys relate.
+The keys are then are sent to the enclave via RPC and processed by:
+- checking the validity of the signature over the viewing key
+- finding the account which has empowered this viewing key.
+We can do that by retrieving the signing public key from the signature.
+By hashing the public key, we can then determine the address of the account.
+- finally the enclave will save the viewing key (which is a public key) against the account, and use it to encrypt any
+sensitive requests (e.g. "eth_call" and "eth_getBalance") that have that address as a "from" field.
```

Comment:
It's not always the `from` field. The logic varies. e.g. for a `eth_getBalance`, we encrypt it with the viewing key associated with the address being queried.